src/Accounts/Authentication/TestCoverage.cs

```diff
@@ -41,36 +44,37 @@ public class TestCoverage : ITestCoverage
             "AzureRM.Storage.ps1"
         };
 
-
-        private static readonly string s_testCoverageRootPath;
+        private static readonly string s_testCoveragePath;
 
         private static readonly ReaderWriterLockSlim s_lock = new ReaderWriterLockSlim();
 
         static TestCoverage()
         {
-
-            var repoRootPath = ProbeRepoDirectory();
-            if (!string.IsNullOrEmpty(repoRootPath))
+            string testCoverageRootPath = null;
+            if (AzureSession.Instance.TryGetComponent<IConfigManager>(nameof(IConfigManager), out var configManager))
             {
-                s_testCoverageRootPath = Path.Combine(repoRootPath, "artifacts", "TestCoverageAnalysis", "Raw");
-                DirectoryInfo rawDir = new DirectoryInfo(s_testCoverageRootPath);
-                if (!rawDir.Exists)
-                {
-                    Directory.CreateDirectory(s_testCoverageRootPath);
-                }
+                testCoverageRootPath = configManager.GetConfigValue<string>(ConfigKeys.TestCoverageLocation);
+            }
+            if (string.IsNullOrEmpty(testCoverageRootPath))
+            {
+                testCoverageRootPath = GetFallbackLocation();
             }
-        }
 
-        private static string ProbeRepoDirectory()
-        {
-            string directoryPath = "..";
-            while (Directory.Exists(directoryPath) && (!Directory.Exists(Path.Combine(directoryPath, "src")) || !Directory.Exists(Path.Combine(directoryPath, "artifacts"))))
+            s_testCoveragePath = Path.Combine(testCoverageRootPath, "TestCoverageAnalysis", "Raw");
+            DirectoryInfo rawDir = new DirectoryInfo(s_testCoveragePath);
+            if (!rawDir.Exists)
             {
-                directoryPath = Path.Combine(directoryPath, "..");
+                Directory.CreateDirectory(s_testCoveragePath);
             }
 
-            string result = Directory.Exists(directoryPath) ? Path.GetFullPath(directoryPath) : null;
-            return result;
+            Console.WriteLine($"Test coverage data location: ${s_testCoveragePath}");
+        }
+
+        private static string GetFallbackLocation()
+        {
+            var profilePath = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+            var azurePath = Path.Combine(profilePath, ".azure");
```

Comment:
It's capitalized A
```suggestion
            var azurePath = Path.Combine(profilePath, ".Azure");
```

Reply:
@isra-fel in my local environments (both Windows and Linux WSL), they are all with lower cases. Do we have any official document mentioning that?

Reply:
No documentation but we use this string for folder name. You should use it too.
https://github.com/Azure/azure-powershell/blob/2d4550c23745a3a9710ecbe5ff3c378f1d4cd300/src/Accounts/Authentication/Properties/Resources.resx#L124

Also I tested cloud shell, our metadata files are all in ".Azure"
![image](https://user-images.githubusercontent.com/11371776/209913134-96415f43-f126-4287-9f33-8e2dce1149ee.png)

Reply:
BTW If you can reproduce how to create the lower case ".azure" folder, it may prove there's an issue in our code.

Reply:
Not able to reproduce. Let me update to upper case first.

Reply:
OK. But I actually suggest using the resource string to avoid magic string, i.e. `Resources.AzureDirectoryName`.

Reply:
Changed to use resource string